### PR TITLE
Cannot easily tell similar named EPeople or Groups apart (usability issue)

### LINKED ISSUE #1072: Cannot easily tell similar named EPeople or Groups apart (usability issue)
**Describe the bug**
In testing other PRs, I've had to recently add some new EPeople & assign them to groups, I've found that the current UI screens rely too much on UUID and Name, making it very difficult to tell similar named EPeople apart.  Examples below...

**To Reproduce**

**Issue 1: Difficult to tell apart two EPeople of same name:**
1. Create two EPerson users with the same name
2. Now, edit a group on the Groups page (http://localhost:4000/admin/access-control/groups)
3. **Usability Issue:** Search for an EPerson with that name...you will not be able to tell the users apart, e.g. in this screenshot there are two "Tim Donohue" Epeople, but no other information provided besides UUID:
two_epeople
4. **Bug:** Even worse, clicking on one of the names brings me to the list of **Groups** instead of to that EPerson.
5. **EXPECTED BEHAVIOR:** We should list an EPerson's email address next to their name (in same column or separate column) in any browse/search results, as an email address is a simple way of distinguishing two people.

**Issue 2: Difficult to tell apart two Groups of a similar name:**
1. Create two Collection Admin groups
2. Now, go to the Groups page (http://localhost:4000/admin/access-control/groups) and try to find the Collection Admin group of a specific Collection.
3. **Usability Issue 1:** You will be unable to tell the Groups apart unless you click on each Collection Admin group one by one.
4. **Usability Issue 2:** Now, click Edit on one of the Collection Admin groups.  It's difficult to tell which Collection it refers to, unless you read the detailed note at the top of the page.
4. **EXPECTED BEHAVIOR 1:** Any Role-based Group should list the name of the Community or Collection next to it (either in the Name column or in a new column) in any browse/search results
5. **EXPECTED BEHAVIOR 2:** When editing a Role-based Group, we should show a Read-only field at the top which lists the Collection or Community name which the Group references.

### PR DESCRIPTION
**Pull request by davidenegretti-4science**
## References
* Fixes #1072 

## Description
- In order to distinguish groups, community or collection is now specified for Role-based Groups
- In order to distinguish EPersons, email address and NetID are now specified

## Instructions for Reviewers

List of changes in this PR:
* A new column with the name of the community/collection has been added to the group list  
  
* A new read-only input field has been added to the "Edit group" page (for role-based groups only)  
  
* A new column with email address and NetID has been added to the EPersons' list  
  

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: src/app/access-control/group-registry/groups-registry.component.html
Comment (tdonohue) [inline src/app/access-control/group-registry/groups-registry.component.html:50]: Instead of making this a `colspan="2"`, we should add a separate `<th>` that says "Collection/Community".  The current display is confusing, as it appears the collection is part of the group "Name" column.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@davidenegretti-4science : Thanks for this contribution!  _Overall, it looks much better and it mostly works._  However, I did run into a few small bugs:
1. For some reason the "Anonymous" group says it's associated with the "Articles" collection.  This is not true, as it's not a `COLLECTION_[uuid]_*` group.  (This bug also appears in your screenshots in the PR description)
2. I found **three places** where your improvements were not yet applied & usability still is bad.
    1. When you search & attempt to add a new EPerson to an existing Group:
        * Go to `/access-control/groups`, click Edit on a Group
        * In the "Add EPeople" section, search for an EPerson.  The results come back **without** the "Identity" column, so it's still hard to tell people apart if they have the same name.
    2. When you search & attempt to add a Subgroup to an existing Group:
        * Go to `/access-control/groups`, click Edit on a Group
        * Scroll down to the "Add Subgroup" search.  Search for a group.  You'll see the results do NOT include the "Collection/Community" column.
     3. When editing an EPerson:
         * Go to `/access-control/epeople`, click Edit on a Person
         * Look at the "Member of these groups:" table.   It does NOT include the "Collection/Community" column.
         
There are also some minor code changes I've requested below after doing a code review.
Overall though, this PR is a big step in the right direction. It just needs some improvements before it can be merged.  Thanks!

**Comment by davidenegretti-4science** [comment]
@tdonohue the first bug seems to be a REST problem for which I just opened a new issue:
[https://github.com/DSpace/DSpace/issues/7928](https://github.com/DSpace/DSpace/issues/7928)

I made al requested code changes and I'll fix the second bug (the missing improvements) ASAP

**Comment by tdonohue** [comment]
Thanks @davidenegretti-4science !  I'll also note that LGTM (which does automated code checks for us) is noting that your code introduces two new alerts for "Unneeded defensive code".   They appear as red boxes on this summary page: https://lgtm.com/projects/g/DSpace/dspace-angular/rev/pr-95ef6962450c9b2d6073093bb5f9cb16c3b2b05a

At a glance, they don't seem directly related to your new code, but may be the result of the code refactors in that `group-form.component.ts` to you added.  If you can find a way to clean them up in the midst of this PR, it'd be appreciated...as we are doing our best to minimize introducing new alerts of this sort.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@davidenegretti-4science : Just flagging this as needs fixes until the missing improvements & LGTM issues are resolved. Let me know when it's ready for review again!

**Comment by davidenegretti-4science** [comment]
LGTM issues should now be resolved, others improvements TBD

**Comment by davidenegretti-4science** [comment]
@tdonohue missing improvements have been completed, pull request is ready for review

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me now @davidenegretti-4science !  I've deployed and tested and the accessibility issues look solved. Requested code updates have also been made (other than the one moved to a separate ticket obviously: DSpace/DSpace#7928).

Thanks!

**Pull request merged**
